pallets/loans-ref/src/lib.rs

```diff
@@ -810,15 +826,11 @@ pub mod pallet {
 							}
 						})?;
 
-					*last_updated = (*last_updated).max(portfolio.last_updated());
-					let old_pv = loan.present_value_at(*last_updated)?;
-
 					let result = f(loan)?;
 
-					*last_updated = Self::now();
-					let new_pv = loan.present_value_at(*last_updated)?;
-
-					Self::update_portfolio_valuation_with_pv(pool_id, portfolio, old_pv, new_pv)?;
+					let new_pv = loan.present_value()?;
+					Self::update_portfolio_valuation_with_pv(pool_id, portfolio, *old_pv, new_pv)?;
+					*old_pv = new_pv;
 
 					Ok((result, active_loans.len().ensure_into()?))
```

Comment:
I've simplified the portfolio changes, now we store the old present value in order not to recompute it again next time.